src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/CraftingRequestResolver.java

```diff
@@ -0,0 +1,164 @@
+    {
+        if (!manager.getColony().getWorld().isRemote)
+        {
+            Colony colony = (Colony) manager.getColony();
+            final ILocation requesterLocation = requestToCheck.getRequester().getRequesterLocation();
+            final AbstractBuilding building = colony.getBuilding(requesterLocation.getInDimensionLocation());
+            if(building instanceof AbstractBuildingWorker)
+            {
+                final ItemStack stack = requestToCheck.getRequest().getResult();
+
+                if (ItemStackUtils.isEmpty(stack))
+                {
+                    return false;
+                }
+
+                if ((((AbstractBuildingWorker) building).canCraft(stack)))
+                {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    @Nullable
+    @Override
+    @SuppressWarnings("squid:LeftCurlyBraceStartLineCheck")
+    /**
+     * Moving the curly braces really makes the code hard to read.
+     */
+    public List<IToken<?>> attemptResolve(
+                                        @NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request)
+    {
+        if (manager.getColony().getWorld().isRemote)
+        {
+            return null;
+        }
+
+        final Colony colony = (Colony) manager.getColony();
+        final ILocation requesterLocation = request.getRequester().getRequesterLocation();
+        final AbstractBuilding building = colony.getBuilding(requesterLocation.getInDimensionLocation());
+        if(canResolve(manager, request) && building instanceof AbstractBuildingWorker)
+        {
+            final ItemStack stack = request.getRequest().getResult();
+            if(ItemStackUtils.isEmpty(stack))
+            {
+                return Lists.newArrayList();
+            }
+
+            final RecipeStorage storage = ((AbstractBuildingWorker) building).getFirstFullFillableRecipe(stack);
+
+            if(storage == null)
+            {
+                return null;
+            }
+            ((AbstractBuildingWorker) building).fullFillRecipe(storage);
+
+            return Lists.newArrayList();
+        }
+
+        return null;
+    }
+
+    @Override
+    public void resolve(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request)
+    {
+        manager.updateRequestState(request.getToken(), RequestState.COMPLETED);
+    }
+
+    @Nullable
+    @Override
+    public IRequest<?> getFollowupRequestForCompletion(
+                                                     @NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> completedRequest)
+    {
+        //No followup needed.
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public IRequest<?> onRequestCancelledOrOverruled(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request)
+    {
+        return null;
```

Comment:
You are going to need to perform clean up.
You might have requested something, if you do not clean up, then we are going to waste stacks.

Reply:
True, definitely